From 8c6869cc21f7bde893ce2ec17e3e23b8998aab5d Mon Sep 17 00:00:00 2001
From: Dilprit Kaur <dilprit007@gmail.com>
Date: Thu, 15 Dec 2022 21:47:44 +0530
Subject: [PATCH] modified functions

---
 client.py | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/client.py b/client.py
index 3fc09b7..443c913 100644
--- a/client.py
+++ b/client.py
@@ -35,11 +35,14 @@ def getDataPoint(quote):
     stock = quote['stock']
     bid_price = float(quote['top_bid']['price'])
     ask_price = float(quote['top_ask']['price'])
-    price = bid_price
+    price = (bid_price+ask_price)/2
     return stock, bid_price, ask_price, price
 
 
 def getRatio(price_a, price_b):
+    if (price_b==0):
+        return
+    return price_a/price_b    
     """ Get ratio of price_a and price_b """
     """ ------------- Update this function ------------- """
     return 1
@@ -48,12 +51,14 @@ def getRatio(price_a, price_b):
 # Main
 if __name__ == "__main__":
     # Query the price once every N seconds.
-    for _ in iter(range(N)):
+    for _ in range(N):
         quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
         """ ----------- Update to get the ratio --------------- """
+        prices ={}
         for quote in quotes:
             stock, bid_price, ask_price, price = getDataPoint(quote)
+            prices[stock]=price
             print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-        print("Ratio %s" % getRatio(price, price))
+        print("Ratio %s" % (getRatio(prices['ABC'], prices['DEF'])))
-- 
2.39.0.windows.1

